### DISCUSSION TITLE
For an app Cask, how to symlink to .app/Contents/MacOS/name  from /usr/local/bin

### DISCUSSION THREAD

**Discussion by benzoid** [Homebrew/discussions#1643] [Casks] [answered]
Hello,
For a Cask with an app install, which essentially seems to copy (rsync/tar?) the Name.app application in the .dmg file to /Applications (with `app "Name.app"`), there is a command line invocation script that is in the app bundle as `Name.app/Contents/MacOS/name`.
I want to add a stage that will create a symbolic link in the user's path (e.g. /usr/local/bin) to that command line invocation script.
I *think* this can be done by adding
```
  postflight do
    Dir.chdir("#{HOMEBREW_PREFIX}/bin") do
      File.symlink("#{appdir}/Name.app/Contents/MacOS/name", "name") if File.exist? "#{appdir}/Name.app/Contents/MacOS/name" unless File.exist? "name"
    end
  end
```
but I'm not very sure.  Does this look right?  What I want to end up with is:

```
% ls -l /usr/local/bin/name
lrwxrwxrwx ... /usr/local/bin/name -> /Applications/Name.app/Contents/MacOS/name
```

**Comment by gromgit** [comment, marked as answer]
https://docs.brew.sh/Cask-Cookbook#stanza-binary

See the `mkvtoolnix` cask for a worked example.

**Reply by benzoid** [reply]
Perfect -- just what I want to do, thanks.
